hadoop-hdds/config/src/main/java/org/apache/hadoop/hdds/conf/ConfigurationSource.java

```diff
@@ -183,6 +183,20 @@ default <T> T getObject(Class<T> configurationClass) {
 
   }
 
+  /**
+   * Returns a singleton instance of the given configuration class.
+   * If an instance of the class has already been created,
+   * it will be returned; otherwise, a new instance will be created,
+   * stored in a map for future retrieval.
+   *
+   * @param configurationClass The class for which a singleton
+   *                           instance is required
+   * @return a singleton instance of the given class
+   */
+  default <T> T getSingletonObject(Class<T> configurationClass) {
+    return getObject(configurationClass);
+  }
+
```

Comment:
`getSingletonObject` has been removed from `ConfigurationSource.java`

Reply:
Thanks @xichen01 for making the change.  On second thought, instead of adding a new method `getSingletonObject`, we should override `getObject` from `ConfigurationSource` with the singleton-based implementation.  This way we can keep passing `ConfigurationSource`.

In the long run, we should move towards passing the type-safe config objects (`DatanodeConfiguration`, `ScmConfig`, etc.)

Reply:
If we change the logic of `ConfigurationSource#getObject` directly, will this cause some of the tests to fail, since at the moment we can assume that each object from `ConfigurationSource#getObject` is independent and unaffected.
If the `ConfigurationSource#getObject` is "Singleton" that maybe cause tests to affect with each other.

Should we keep this implemented(`getSingletonObject`) temporarily, and implement a "Singleton" `ConfigurationSource#getObject` in another MR?